### DISCUSSION TITLE
power_outage_count for battery powered devices?

### DISCUSSION THREAD

**Discussion by michal-bartak** [Koenkk/zigbee2mqtt#13018] [Devices]
After updating to 1.26.0 I can see that power_outage_count is exposed for batter Aqara door & window contact sensor (https://www.zigbee2mqtt.io/devices/MCCGQ11LM.html#xiaomi-mccgq11lm)

I can see documentation has been updated too. But what's the point of measuring power outage for battery-powered devices?
Maybe in this case it measures deep-sleep or something. But still... what's the point?

Also not sure the use of `device_temperature` instead of `temperature` is correct choice. I understand that some devices provide monitoring of internal temperature in contrary to measuring an environment temperature. But in case of this device it's opposite IMO.

**Comment by sjorge** [comment]
I also noticed this https://github.com/Koenkk/zigbee2mqtt/issues/12986

It seems to measure the times the device has been without power, although in my case it takes a significant time for the device to actually be drained of the remaining power in the circuit unless I short the pins.

I too don't really see the point of tracking this but if some people want it, I just added `power_outage_count` to [filtered_attributes](https://www.zigbee2mqtt.io/guide/configuration/devices-groups.html#common-device-options) and I no longer see or have to deal with it.

**Reply by michal-bartak** [reply]
Thanx for the suggestion of using filtered attributes.
While it (likely) doesn't publish data for filtered attributes, it still creates sensor in HomeAssistant :(

**Reply by sjorge** [reply]
I wasn't aware of the later, as I do not use HomeAssistant.

Seems there was a feature request to filter those out of expose that was never implemented, I think the HA sensors are based on the expose data. https://github.com/Koenkk/zigbee2mqtt/issues/9635

So I don't think there is a way to remove them for HA for now.

**Comment by Scoff123** [comment]
I just came across this discussion after searching for my device model and power_outage_count. Having just received 2 Xiaomi smart switches, model WXKG11LM I immediately paired them to my controller in Zigbee2MQTT. After they paired I could see the power_outage_count for each device - the first one showed a count of 61 and the second switch showed a count of 24. Made me curious as to what had generated these counts, and also, Z2M defines this data as "Number of power outages (since last pairing)". That can't be correct, as surely upon pairing to my network, the count would be 0, as I obviously haven't removed the battery since pairing them?

**Reply by pqpxo** [reply]
I can see the same thing in regards to the large power_outage_count. 

I've just added a Aqara Door/Window Contact Sensor and the count is already '119'

**Reply by michal-bartak** [reply]
Couldn't it be a number of deep-sleeps?

**Comment by JonSilver** [comment]
It's simple: it's the number of brown-outs. Alas the battery voltage measurement on many of these sensors isn't a great guide to a dying battery. When the device tries to power up and do something that draws a lot of current (comparatively), the voltage will drop off a cliff all of a sudden and the processor will suffer a brown-out. When it resets, it'll note the reason for the reset, and count the brown-out. The battery voltage might still read as healthy under a light load, but actually very few of these devices ever read their battery voltage particularly accurately anyway.

So... a rapidly rising power outage count is a warning that the battery needs replacing. Shame it's so hard to monitor rapid rises in this sort of sensor in platforms like HA.

**Reply by Koenkk** [reply]
Learned something today, thanks! 😄

**Reply by sjorge** [reply]
Also at least the Aqara TVOC seems to lose binding and reporting when that happens. It’s on my todo to find a way to reapply those when that happens.

Sadly there is no deviceAnnounce eveny when it recovers. It also seems to sometimes reset when the coordinator is ofline for a few minutes. Probably the device trying to power cycle to hope to recover the connection?

**Reply by JonSilver** [reply]
> Also at least the Aqara TVOC seems to lose binding and reporting when that happens. It’s on my todo to find a way to reapply those when that happens.
> 
> 
> 
> Sadly there is no deviceAnnounce eveny when it recovers. It also seems to sometimes reset when the coordinator is ofline for a few minutes. Probably the device trying to power cycle to hope to recover the connection?

The binding phase is the most current-demanding part of a sensor's life, requiring it to power up its radio, transmit its details and wait for responses and configuration from the controller. Even if the designers were very clever about it, that's orders of magnitude more power consumption all in one go than at any other time. The rest of its life it's mostly in deep sleep, waking occasionally to read its sensors and transmit a small packet of data. But eventually even the power required for that will catch up with a depleted battery.

In my experience most Aqara sensors with a high power outage count will just reconnect to the controller once their battery has been changed and their button pressed once.

**Reply by sjorge** [reply]
Even with 100% fresh batteries (tested with a multi meter) the device starts power cycling once every ~18h for me if the coordinator is offline. It connects back after such a power cycle if the coordinator is online then. I think it's just part of the 'I lost my connection and cannot restore it' handling. The main issue is that when it does that it doesn't report anymore and the bindings are gone (e.g. if you bind it to a TRV). It's not really a rapid increase in that case as you can roughly predict amount the number has gone up by how long the coordinator has been offline.

In general usage with healthy batteries it doesn't really go up, but I had it try to send a message just as I was rebooting the coordinator and have it go offline until it power cycles later. It's easy to reproduce on a dev network though. 

Overall I am not happy with this specific aqara device, but I'm not happy with the develco voc sensor either which I am trying to find a replacement for. The develco's all die after ~8-12 months for me, yet there non VOC sensors are all rock solid.

But if the Aqara one would not lose it's bindings it would at least be usable.

**Reply by JonSilver** [reply]
Interesting. So when it does this, are you seeing specifically the Power Outage Count increase? And by how much each time?

Incidentally, a weak, depleted lithium coin-cell battery can still show full voltage when under very low current draw. I've just checked all the ones in my dead battery box (awaiting a trip to the recycling centre) and all the coin cells gave between 3V and 3.1V apart from one which was sitting at 2.4V. But if I put these in even the most low-powered device with a radio on board, it wouldn't work for long if at all. Hence, they're dead to me. 😊

**Reply by sjorge** [reply]
Roughly 1 every 18h ish if the coordinator is down. It seems to completely stop after a few days and doesn’t try to reconnect at all. Need a lot more data but currently it looks like:

1. lose connection (coordinator down)
2. ~18h outage count +1 (tries to reconnect)
3. (Coordinator still down) ~18h count +1 (tries recommect)
4. Same as 3., but it stops retrying


It’s very hard to get reliable data

**Reply by JonSilver** [reply]
Where is this sensor situated?

Another thing I forgot to mention: temperature. Batteries' chemistry slows down dramatically when it's cold, and as this happens their ability to supply charge quickly aka supplying current drops off sharply. Then you take the coin cell out of the sensor, and it warms rapidly in your hand, then your low-load meter says it's fine, so you pop it back in the sensor and it fails again.

The batteries supplied ready-fitted in many sensors purchased on AliExpress etc are often inferior cells and quite old. They fail quickly and I'm convinced that many of the problems people experience with unreliable or difficult Zigbee sensors are actually down to this problem. Sometimes they won't bind to the controller. Sometimes they go dead after a short time. Or when, for example, a contact sensor is fitted to a draughty external door and the already weak battery just gets really cold. 😊

**Reply by sjorge** [reply]
Mine is indoor around 18-20C and I was using a fresh set of varta cells. As long as it's on the netwerk it generally works fine.
I'm pretty sure there is this weird re-connect logic that is specific to these devices, they also never send a deviceAnnounce when rejoining which is also not great.

**Comment by dersimn** [comment]
For me it's just a metric to detect a dying battery.. this sensor shows 2.9V; 77% battery left, but see the orange dots of motion reportings vs. `power_outage_count`:

<img width="543" height="394" alt="Screenshot 2026-03-30 at 20 21 37" src="https://github.com/user-attachments/assets/7c54b74e-7e79-4432-b954-2c30c0ef05e2" />
